sockets/unix_socket.go

```diff
@@ -79,8 +79,14 @@ func WithChmod(mask os.FileMode) SockOption {
 
 // NewUnixSocketWithOpts creates a unix socket with the specified options
 func NewUnixSocketWithOpts(path string, opts ...SockOption) (net.Listener, error) {
-	if err := os.Remove(path); err != nil && !os.IsNotExist(err) {
-		return nil, err
+	// Using syscall.Unlink(), not os.Remove() to prevent deleting the socket if it's in use
+	if err := syscall.Unlink(path); err != nil && !os.IsNotExist(err) {
+		if err != syscall.EISDIR {
```

Comment:
> OS X and Linux differ on whether unlink(dir)
> returns EISDIR,

https://golang.org/src/os/file_unix.go#L289

Reply:
Ah, interesting. I guess it's still ok to retry on Linux only then? Added a comment

Reply:
Also added a test